src/main/java/command/Command.java

```diff
@@ -0,0 +1,195 @@
+
+
+import ui.Ui;
+
+import java.io.IOException;
+
+/**
+ * Command class responsible for executing different command
+ *
+ * @author Kan Jitpakdi
+ * @author GitHub: kanjitp
+ * @version 0.01
+ * @since 0.00
+ */
+public class Command {
+    /**
+     * Different type of Command available
+     */
+    public enum CommandType {
+        TODO, DEADLINE, EVENT, LIST, DATE, FIND, DONE, DELETE, COMMANDS, BYE
+    }
+
+    private String fullCommand;
+    private CommandType commandType;
+    private boolean isExit;
+
+    /**
+     * Default constructor for Command.
+     * Create a Command object from given full Command
+     *
+     * @param fullCommand command with its arguments and tagger
+     * @throws AliceException invalid argument for command
+     */
+    public Command(String fullCommand) throws AliceException {
+        this.fullCommand = fullCommand;
+        commandType = Parser.stringToCommand(fullCommand.split(" ")[0]);
+        isExit = commandType == CommandType.BYE;
+    }
+
+    /**
+     * Return whether the command given is of type Bye
+     *
+     * @return if the command is bye or not
+     */
+    public boolean isExit() {
+        return this.isExit;
+    }
+
+    /**
+     * Execute the command according to its type
+     *
+     * @param taskDialog taskDialog for the command to interact with
+     * @param storage    the storage for the command to interact with
+     * @throws DialogException dialog cannot have the same id while the app is running
+     * @throws IOException     if there is any error dealing with the system IO
+     */
+    public void execute(TaskDialog taskDialog, Storage storage) throws DialogException, IOException {
+        switch (commandType) {
+        case LIST:
+            System.out.println(taskDialog);
+            break;
+        case DATE:
+            try {
+                if (fullCommand.split(" ").length == 1) {
+                    throw new EmptyDescriptionException("The date of deadline cannot be empty.");
+                }
+                String dlString = fullCommand.substring(("date ").length());
+                System.out.println(taskDialog.by(dlString));
+            } catch (EmptyDescriptionException | EmptyTaggerException | InvalidTimeFormatException e) {
+                Ui.printError(e);
+            }
+            break;
+        case FIND:
+            try {
+                if (fullCommand.split(" ").length == 1) {
+                    throw new EmptyDescriptionException("The keyword of find cannot be empty. Try using command " +
+                            "'list' if you want to see the full list");
+                }
+                String kwString = fullCommand.substring(("find ").length());
+                System.out.println(taskDialog.with(kwString));
+            } catch (EmptyDescriptionException | EmptyTaggerException | InvalidTimeFormatException e) {
+                Ui.printError(e);
+            }
+            break;
+
+        case TODO: {
```

Comment:
Are we supposed to put case statements in curly brackets in switch statements?

Reply:
Fixed thanks a lot :))